packages/next/src/server/image-optimizer.ts

```diff
@@ -438,6 +451,21 @@ export function getMaxAge(str: string | null | undefined): number {
   return 0
 }
 
+export function shouldUsePreviouslyCachedEntry(
+  upstreamImage: ImageUpstream,
+  previousCacheEntry: IncrementalCacheItem | undefined
+): previousCacheEntry is NonNullable<IncrementalCacheItem> & {
+  value: CachedImageValue
+} {
+  return (
+    previousCacheEntry?.value?.kind === 'IMAGE' &&
+    // If the cached image is optimized
+    previousCacheEntry.value.upstreamEtag !== previousCacheEntry.value.etag &&
```

Comment:
Can you explain why upstreamEtag should not be equal to etag?

This seems like it will always return true when not equal and therefore it will never actually revalidate, which seems wrong to me.

Reply:
What this PR tries to do is prevent hitting [this specific path](https://github.com/vercel/next.js/pull/67257/files/a1f627d30eed6685c694b13c02ae7eb894fda56f#diff-07cdc7c577b57eb1e3cf336ecca5995abae754a39c20f5bba67794dece3ddf21R690-R718), with `optimizeImage` is the expensive operation.

`upstreamEtag` represents the original source image's etag (currently calculated via the same function `getImageEtag`), whereas the cache entry's `etag` represents the resulting image's `etag`, i.e. in the case image optimization is done, the `etag` will be the hash of that optimized image instead of the upstream image.

This condition exists to cause a revalidation on cases where the optimization did not happen previously (in this case we fallback to using `upstreamBuffer` and `upstreamEtag` for the image), so this will fail early on those cases and cause a new optimization attempt. Although now thinking back at it the likelyhood of the image optimization ending up with the same outcome is quite likely anyways.

When this condition is true, we'll check the previous upstream image's etag against the current upstream image's etag to determine if we should revalidate or not.

Reply:
The optimization not happening could be because of a fallback (such as an error was thrown) or it could be because the image was vector form (svg doesnt need to be optimized). 

Either way, we still need to revalidate those images because the upstream response could have changed. So I suggest removing line 463 and that way the following line can actually compare the new upstream etag.

Reply:
Hmm, I'm not quite following, as the function is `shouldUsePreviouslyCachedEntry`, all of the conditions inside it need to return true for the revalidation not to occur.

This is how I see it:
- If this comparison is false, we will revalidate the image
- If this comparison is true, we will do the etag comparison (line 465) to determine if we want to revalidate the image.

Am I misunderstanding something? What do you mean by revalidation? I'm fine with removing this line, but don't fully understand the reasoning 😅

Reply:
So what is the scenario where the line is needed? Is there a test for this?

I can't see your link anymore because you force pushed.

(note: you don't need to force push because we squash on merging)

Reply:
Oh I see now. I read through this again and I see that the line is handling the case when svg or animated. I think the comment should mention this.

Reply:
Added a more clarifying comment for the reasoning 👍